src/cache/cache_las.c

```diff
@@ -1186,72 +1194,102 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
-             *	2. durable timestamp as not max timestamp.
-             *	3. for an in-progress prepared update, durable
-             *	timestamp should be zero.
-             *	4. no restriction on durable timestamp value
-             *	for other updates.
-             */
-            WT_ASSERT(session, prepare_state != WT_PREPARE_LOCKED &&
-                durable_timestamp != WT_TS_MAX &&
-                (prepare_state != WT_PREPARE_INPROGRESS || durable_timestamp == 0));
+        /* Remove the obsolete entries */
+        if (remove_entry) {
+            /* Remove each individual LAS record till the current record. */
+            for (i = 0; i < skip_update_modify_cnt + 1; i++) {
+                WT_ERR(cursor->prev(cursor));
 
-            WT_ASSERT(session,
-              (prepare_state == WT_PREPARE_INPROGRESS || durable_timestamp >= las_timestamp));
+                if (WT_VERBOSE_ISSET(session, WT_VERB_LOOKASIDE_ACTIVITY)) {
+                    WT_ERR(
+                      cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+                    WT_ERR(cursor->get_value(
+                      cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+
+                    __wt_verbose(session, WT_VERB_LOOKASIDE_ACTIVITY,
+                      "Sweep removing lookaside entry with "
+                      "btree ID: %" PRIu32 " key size: %" WT_SIZET_FMT " prepared state: %" PRIu8
+                      " record type: %" PRIu8 " durable timestamp: %" PRIu64
+                      " transaction ID: %" PRIu64,
+                      las_btree_id, las_key.size, prepare_state, upd_type, durable_timestamp,
+                      las_txnid);
+                }
 
-            /*
-             * There are several conditions that need to be met
-             * before we choose to remove a key block:
-             *  * The entries were written with skew newest.
-             *    Indicated by the first entry being a birthmark.
-             *  * The first entry is globally visible.
-             *  * The entry wasn't from a prepared transaction.
-             */
-            if (upd_type == WT_UPDATE_BIRTHMARK &&
-              __wt_txn_visible_all(session, las_txnid, durable_timestamp) &&
-              prepare_state != WT_PREPARE_INPROGRESS)
-                removing_key_block = true;
-            else
-                removing_key_block = false;
+                WT_ERR(cursor->remove(cursor));
+                ++remove_cnt;
+            }
+
+            WT_ERR(cursor->next(cursor));
+
+            /* Get the LAS key and value in case if their replaced. */
+            if (WT_VERBOSE_ISSET(session, WT_VERB_LOOKASIDE_ACTIVITY)) {
+                WT_ERR(
+                  cursor->get_key(cursor, &las_btree_id, &las_key, &las_timestamp, &las_txnid));
+                WT_ERR(cursor->get_value(
+                  cursor, &durable_timestamp, &prepare_state, &upd_type, &las_value));
+            }
```

Comment:
I'm confused why this code is here? It doesn't seem like the retrieved key/value information is ever used?

Reply:
> I think there still may be something here, because the err label is going to free sweep_key. Should we jump to srch_notfound instead

Once we reach the limit, we just break. In that code flow, we didn't free the `sweep_key` because the logic of `sweep_key` free is in `if (0)` condition.

> I'm confused why this code is here? It doesn't seem like the retrieved key/value information is ever used?

I remove the above statements with new logic.